shared/test/test_sound_library_api.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'test_helper'
+require 'sound_library_api'
+
+class SoundLibraryTest < Minitest::Test
+  include Rack::Test::Methods
+  include SetupTest
+
+  def build_rack_mock_session
+    @session = Rack::MockSession.new(SoundLibraryApi, 'studio.code.org')
+  end
+
+  def test_get_sound
+    stubs(:cache_for).returns(nil)
+    get '/api/v1/sound-library/test_sound.mp3'
+    assert last_response.ok?
+  end
+
+  def test_get_deleted_sound
+    stubs(:cache_for).returns(nil)
+    get '/api/v1/sound-library/test_sound.mp3'
+    assert last_response.ok?
```

Comment:
At the very least, this test will fail if someone changes the content or ordering of the S3 requests that API generates.  In this case, because the recorded responses show the first version is a delete marker, the API continues and requests the second version - if it failed to do so, the test would fail even though we've recorded the responses.